includes/functions/plugins/article-export.php

```diff
@@ -384,27 +391,110 @@ function get_file_sets() {
 	}
 
 	function output_zip() {
+		// prepare folder
+		$tmp_folder = 'export_xml_' . get_current_user_id();
+
+		if ( ! is_dir( get_temp_dir() . $tmp_folder ) ) {
+			mkdir( get_temp_dir() . $tmp_folder );
+		}
+
+		// create zip
+		$tmp_zip_file = get_temp_dir() . $tmp_folder . '/output.zip';
+		if ( file_exists( $tmp_zip_file ) ) {
+			unlink( $tmp_zip_file );
+		}
+		$zip  = new \ZipArchive( $this->print_issue_title );
+		$code = $zip->open( $tmp_zip_file, \ZipArchive::CREATE | \ZipArchive::OVERWRITE );
+
 		$file_sets = $this->get_file_sets();
-		$zip = new Articles_Zip( $this->print_issue_title );
 
+		$sub_folders = array();
 		foreach ( $file_sets as $article_id => $files ) {
+			foreach ( $files as $file ) {
+				if ( stripos( $file->filename, '.xml' ) !== false ) {
+					$filename                   = explode( '.', $file->filename );
+					$sub_folders[ $article_id ] = $filename[0];
+					break;
+				}
+			}
 
-			//force an array
-			$files = is_array( $files ) ? $files : [ $files ];
-			$zip->addFiles( $files );
+			if ( ! $sub_folders[ $article_id ] ) {
+				$sub_folders[ $article_id ] = $article_id;
+			}
 		}
 
-		$zip->output_zip();
+		// add xml files
+		foreach ( $file_sets as $article_id => $files ) {
+
+			$zip->addEmptyDir( $sub_folders[ $article_id ] );
+
+			// force an array
+			$files = is_array( $files ) ? $files : array( $files );
+			foreach ( $files as $file ) {
+				$zip->addFromString( $sub_folders[ $article_id ] . '/' . $file->filename, $file->contents );
+			}
+		}
+
+		// add images (by path)
+		if ( is_array( $this->images ) ) {
+			foreach ( $this->images as $article_id => $images ) {
+				foreach ( $images as $image_full_path => $image_name ) {
+					$zip->addFile( $image_full_path, $sub_folders[ $article_id ] . '/' . remove_accents( $image_name ) );
+				}
+			}
+		}
+
+		$zip->close();
+
+		$this->out_zip_file( $tmp_zip_file );
+	}
+
+	/**
+	 * Builds the file name for the zip
+	 * Uses the print issue title & day/time
+	 *
+	 * @uses get_timezone
```

Comment:
Doesn't seem like this comment is accurate as this method isn't used here but in `get_zip_file_name`

Reply:
@dkotter This was actually refactored and now lives [here](https://github.com/10up/eight-day-week/blob/950ced938e388276dad7db0cfb1fb8d92e862a0e/includes/lib/class-articles-zip.php#L124) with a more accurate comment.

Reply:
Updated